python_scripts/ensemble.py

```diff
@@ -923,15 +933,16 @@ def plot_decision_function(X, y, clf, sample_weight=None, fit=True, ax=None):
 #
 # ### Gradient-boosting decision tree
 #
-# For gradient-boosting, parameter tuning is a combination of several
-# parameters instead of setting one after the other each parameter. The
+# For gradient-boosting, parameters are coupled, so we can not anymore
+# set the parameters one after the other. The
```

Comment:
```suggestion
# For gradient-boosting, parameters are coupled, so we can not
# set the parameters sequentially one after the other anymore. The
```

Reply:
I understand now, we can't separately tune the parameters of the estimators because the estimators are trained sequentially here. Not sure of the best way to say this but maybe we could make more explicit it is due to the 'coupling'/'connection' (??) of the estimators?

Reply:
My understanding is that 'learning_rate' is *coupled* with 'max_depth'. I cannot tune one without tuning the other
 Like it is said below I think.

I think we still need to improves on it...

Reply:
Ahhh I see. I am confused about the 'anymore'. Previously in this notebook we tune a decision tree:
```
param_grid = {
    "max_depth": [3, 5, 8, None],
    "min_samples_split": [2, 10, 30, 50],
    "min_samples_leaf": [0.01, 0.05, 0.1, 1],
}
```
but these parameters would be somewhat linked as well? Maybe we should discuss this.